aiidalab_widgets_base/structures.py

```diff
@@ -735,6 +738,40 @@ def _default_structure(self):
         return None
 
 
+def _register_operator(operator):
+    """
+    check if the operation can process.
+    If not valid, the decorated method should not be executed but
+    pop up a warning message asking for doing the prerequisites.
+    """
+
+    @functools.wraps(operator)
+    def inner(ref, *args, **kwargs):
+
+        if not ref.structure:
+            ref._status_message.message = """
+            <div class="alert alert-info">
+            <strong>Please choose a structure first.</strong>
+            </div>
+            """
+        elif not ref.selection:
+            ref._status_message.message = """
+            <div class="alert alert-info">
+            <strong>Please select atoms first.</strong>
```

Comment:
```suggestion
            <strong>Please make a selection first.</strong>
```
They don't necessarily have to be atoms, do they?

Reply:
My point is that we could potentially have "structures" that are not made up of "atoms", e.g., molecular nano structures. So by avoiding that term, it's just as correct, but less restrictive.

Reply:
> My point is that we could potentially have "structures" that are not made up of "atoms", e.g., molecular nano structures

But do we support these kind structures in this widget?

Reply:
I withdraw my suggestion.